controlplane/internal/upgrade/upgrade_test.go

```diff
@@ -186,6 +188,121 @@ var _ = Describe("OpenShift Upgrader", func() {
 	})
 })
 
+type UpdateHistory struct {
+	History []configv1.UpdateHistory `yaml:"history"`
+}
+
+var _ = Describe("detect upgrade status", func() {
+	var (
+		ctx             context.Context
+		mockCtrl        *gomock.Controller
+		mockRemoteImage *containers.MockRemoteImage
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockRemoteImage = containers.NewMockRemoteImage(mockCtrl)
+
+	})
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+	DescribeTable("detect upgrade status", func(historyYAML string, isUpgradeExpected bool) {
+		updateHistory := &UpdateHistory{}
+		Expect(yaml.Unmarshal([]byte(historyYAML), updateHistory)).To(Succeed())
+
+		clusterVersion := getClusterVersion(updateHistory.History)
+
+		fakeClient := fake.NewClientBuilder().
+			WithScheme(testScheme).
+			WithObjects(&clusterVersion).
+			WithStatusSubresource(&configv1.ClusterVersion{}).
+			Build()
+
+		upgrader := upgrade.NewOpenshiftUpgrader(fakeClient, mockRemoteImage)
+
+		inProgress, err := upgrader.IsUpgradeInProgress(ctx)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(inProgress).To(Equal(isUpgradeExpected))
+	},
+		Entry("regular upgrade completed",
+			`
+    history:
+    - acceptedRisks: 'Precondition "ClusterVersionRecommendedUpdate" failed because
+        of "UnknownUpdate": RetrievedUpdates=False (VersionNotFound), so the recommended
+        status of updating from 4.19.0-ec.2 to 4.19.0-ec.5 is unknown.'
+      completionTime: "2025-06-16T11:43:20Z"
+      image: quay.io/openshift-release-dev/ocp-release@sha256:fde51d1c35425759c83ea0895d0d4fc0f88f6a49a0d080207f15a7bbe3e9982c
+      startedTime: "2025-06-16T10:40:09Z"
+      state: Completed
+      verified: true
+      version: 4.19.0-ec.5
```

Comment:
Changes look fine but I wonder if it would be easier and more maintainable to add new tests using the configv1.UpdateHistory struct rather than a yaml string? 
 ```suggestion
    history: []configv1.UpdateHistory{
			{
				State:   configv1.CompletedUpdate,
				Version: "4.19.0-ec.5",
				CompletionTime: "2025-06-16T11:43:20Z",
				Verified: true,
			},
		}

```

Reply:
dates would be something like `meta1v1.Date(2025, 6, 16, 13, 26, 0, UTC)`